apps/src/code-studio/components/libraries/LibraryCreationDialog.jsx

```diff
@@ -81,14 +84,46 @@ class LibraryCreationDialog extends React.Component {
       this.state.libraryClientApi,
       error =>
         this.setState({dialogState: DialogState.ERROR, errorMessage: error}),
-      libraryDetails =>
-        this.setState({
-          dialogState: DialogState.DONE_LOADING,
-          libraryDetails: libraryDetails
-        })
+      this.onLibraryLoaded
     );
   };
 
+  onLibraryLoaded = async libraryDetails => {
+    const defaultNewState = {
+      dialogState: DialogState.DONE_LOADING,
+      libraryDetails
+    };
+
+    try {
+      const profaneWords = await this.findProfanity(
+        libraryDetails.librarySource
+      );
+      if (profaneWords) {
+        this.setState({
+          dialogState: DialogState.CODE_PROFANITY,
+          errorMessage: i18n.libraryCodeProfanity({
+            profanityCount: profaneWords.length,
+            profaneWords: profaneWords.join(', ')
+          })
+        });
+      } else {
+        this.setState(defaultNewState);
+      }
+    } catch {
+      // Still show dialog content if request errors
+      this.setState(defaultNewState);
+    }
+  };
+
+  findProfanity = text => {
+    return $.ajax({
+      url: '/profanity/find',
+      method: 'POST',
+      contentType: 'application/json;charset=UTF-8',
+      data: JSON.stringify({text})
+    });
+  };
```

Comment:
`findProfanity` doesn't have any particular dependencies on this instance, or even this component.  Instead of making it an instance method and passing it through as a prop to `LibraryPublisher`, you could define it in a static context and import it separately in both components.